src/api/courses.js

```diff
@@ -4,13 +4,93 @@ const router = require('express').Router({
 
 const { check } = require('express-validator/check')
 const { matchedData } = require('express-validator/filter')
+const moment = require('moment')
 
-const { Course, Queue, User } = require('../models')
+const { Course, Queue, Question, User, Sequelize } = require('../models')
 const { requireCourse, requireUser, failIfErrors } = require('./util')
 const requireAdmin = require('../middleware/requireAdmin')
 const requireCourseStaff = require('../middleware/requireCourseStaff')
 const safeAsync = require('../middleware/safeAsync')
 
+/**
+ * Helper for getting course question data
+ * This filters all present columns in question data returned from the sequelize query to
+ * include only the columns in our list of selectedColumns.
+ * This is to account for the case where a column may be absent or missing from questions.
+ */
+
+const getColumns = questions => {
+  const columns = new Set()
+  const selectedColumns = new Set([
+    'id',
+    'topic',
+    'enqueueTime',
+    'dequeueTime',
+    'answerStartTime',
+    'answerFinishTime',
+    'comments',
+    'preparedness',
+    'askeBy.UserLocation',
+    'answeredBy.AnsweredBy_netid',
+    'answeredBy.AnsweredBy_UniversityName',
+    'askedBy.AskedBy_netid',
+    'askedBy.AskedBy_UniversityName',
+    'queue.queueId',
+    'queue.courseId',
+    'queue.QueueName',
+    'queue.QueueLocation',
+    'queue.Queue_CreatedAt',
+    'queue.course.CourseName',
+  ])
+
+  questions.forEach(question => {
+    Object.keys(question).forEach(questionKey => {
+      if (selectedColumns.has(questionKey)) {
+        columns.add(questionKey)
+      }
+    })
+  })
+  return columns
+}
+
+const getCsv = questions => {
+  const columns = getColumns(questions)
+  // Taken from https://stackoverflow.com/questions/8847766/how-to-convert-json-to-csv-format-and-store-in-a-variable
+  const timeFields = new Set([
+    'queue.Queue_CreatedAt',
+    'enqueueTime',
+    'dequeueTime',
+    'answerStartTime',
+    'answerFinishTime',
+  ])
+  const header = Array.from(columns)
+  const replacer = (key, value) => (value === null ? '' : value)
+  const csv = questions.map(row =>
+    header
+      .map(fieldName => {
+        if (timeFields.has(fieldName)) {
+          const time = row[fieldName]
+          const formattedTime =
+            time !== null
+              ? moment
+                  .tz(time, 'YYYY-MM-DD HH:mm:ss.SSS Z', 'US/Central')
+                  .format('YYYY-MM-DD HH:mm:ss')
+              : ''
+          return JSON.stringify(formattedTime, replacer)
+        }
+        return JSON.stringify(row[fieldName], replacer)
+      })
+      .join(',')
+  )
+  const splitHeader = header.map(h => {
+    const headerSplit = h.split('.')
+    return headerSplit[headerSplit.length - 1]
+  })
+  csv.unshift(splitHeader.join(','))
+
+  return csv.join('\r\n')
```

Comment:
Is there a particular reason this is using `\r\n` and not just `\n`? Would be good to have that reasoning documented inline.